state/redis/redis.go

```diff
@@ -162,7 +160,11 @@ func (r *StateStore) Init(ctx context.Context, metadata state.Metadata) error {
 
 // Features returns the features available in this state store.
 func (r *StateStore) Features() []state.Feature {
-	return r.features
+	if r.clientHasJSON {
```

Comment:
By doing this, we make the component report that query API is supported only if it's actually supported - if it's connected to a Redis that has RedisJSON

This was needed for the conformance tests too